manager/manager.go

```diff
@@ -187,6 +191,44 @@ func (rm *realManager) housekeep(start, end time.Time) {
 	}
 }
 
+// Map infra containers to pods - only if we have any pods of course.
+// Clunky but required without significant API changes (coming soon!).
+func (rm *realManager) mapInfraContainersToPods(sd *syncData) {
+	if len(sd.data.Pods) > 0 {
+		sd.mutex.Lock()
+		defer sd.mutex.Unlock()
+
+		// Iterate slice in reverse as we are going to be deleting containers if we manage to match infra containers.
+		for containerIndex := len(sd.data.Containers) - 1; containerIndex >= 0; containerIndex-- {
+			cont := sd.data.Containers[containerIndex]
+			podName, podNameFound := cont.Spec.Labels[datasource.KubernetesPodNameLabel]
+			podNamespace, podNamespaceFound := cont.Spec.Labels[datasource.KubernetesPodNamespaceLabel]
+
+			// Not a pod container then continue.
+			if !podNameFound || !podNamespaceFound {
+				continue
+			}
+
+			// Find a pod with these details.
+			for podIndex := range sd.data.Pods {
+				pod := &sd.data.Pods[podIndex]
+				// If we find a matching pod then add the container to the pod's containers slice.
+				if pod.Name == podName && pod.Namespace == podNamespace {
+					cont.Hostname = pod.Hostname
```

Comment:
Bit confused by this. We're adding the infra container to the appropriate pod so this is done.